asq-exercise-progress.html

```diff
@@ -0,0 +1,310 @@
+      this._notifyPolymer();
+    },
+
+    _userBlurred: function(user) {
+      this._makeSureUserExists(user);
+    },
+
+    _resetUser: function(user) {
+      this._users[user.id].submitted = false;
+      this._users[user.id].working = false;
+      this._users[user.id].focused = false;
+      this._users[user.id].idle = false;
+    },
+
+    _userSubmitted: function(user) {
+      this._makeSureUserExists(user);
+      this._resetUser(user);
+      this._users[user.id].submitted = true;
+      this._submissionCounter++;
+    },
+
+    _userConnected: function() {
+      this.connectedViewersNum++;
+    },
+
+    _userDisconnected: function() {
+      this.connectedViewersNum--;
+    },
+
+    _notifyPolymer: function() {
+      _users = this._users;
+      this._users = {};
+      this._users = _users;
+    },
+
+    _setVelocity:  function(property, velocity) {
+      let that = this;
+      this.customStyle[property] = velocity + 's';
+      this.updateStyles();
+      setTimeout(function() {
+        that.customStyle[property] = '0s';
+        that.updateStyles();
+      }, velocity * 1000);
+    },
+
+    _visualizeChange: function() {
+      let that = this;
+
+      setInterval(function() {
+        let submissionChange = that._calcChange(that._submissionCounter);
+        let workingChange = that._calcChange(that._workingCounter);
+        let focusedChange = that._calcChange(that._focusedCounter);
+        let idleChange = that._calcChange(that._idleCounter);
+
+        that._setVelocity('--submitted-bar-duration', that._determineVelocity(submissionChange));
+        that._setVelocity('--working-bar-duration',  that._determineVelocity(workingChange));
+        that._setVelocity('--focused-bar-duration',  that._determineVelocity(focusedChange));
+        that._setVelocity('--idle-bar-duration',  that._determineVelocity(idleChange));
+
+        that._submissionCounter = 0;
+        that._workingCounter = 0;
+        that._focusedCounter = 0;
+        that._idleCounter = 0;
+      }, 5000);
+    },
+
+    _calcChange: function(num) {
+      let total = this.submissionsNum + this._workingUsersNum + this._focusedUsersNum + this._idleUsersNum;
+      return num/total * 100;
+    },
+
+    _determineVelocity: function(num) {
+      if (num == 0) return 0;
+      else if(num < 5) return 10;
+      else if(num < 15) return 8;
+      else if(num < 30) return 6;
+      else if(num < 50) return 4;
+      else if(num < 75) return 2;
+      else if(num <= 100) return 1;
+      else return 0 // change was higher than 100%
+    },
+
+    created: function() {
+      document.addEventListener('asq-ready', function(evt) {
+        try {
+          this._subscribeToEvents(evt.detail.asqEventBus);
+        } catch (err) {
+          console.debug('failed to _subscribeToEvents');
+        }
+      }.bind(this));
+      this._visualizeChange();
+    },
+
+    _subscribeToEvents: function(eventBus) {
+      let that = this;
```

Comment:
instead of `that` you can bind the event hanlder function to `this` like so:

```js
eventBus.on('asq:sessionEventBeamer', function(evt) {

  const user = evt.data.sessionEvent.user;
  switch (evt.data.sessionEvent.type) {
    case 'folo-connected':
    case 'ghost-connected':
      return this._userConnected();
    //rest of your code
  }
}.bind(this))
```